### ISSUE TITLE
Install as project dependency?

### ISSUE THREAD

**Issue by jehoshua02** [GitbookIO/gitbook#554]
I know the gitbook docs instruct to install gitbook globally, but is it possible to install it as a project dependency? You don't want to write the book against one version of gitbook and build with a non-backward compatible version later. Installing it globally, other developers have no idea what version of gitbook was originally used. Am I missing something?

**Comment by AaronO** [comment]
@jehoshua02 We try to be as backwards compatible as possible.

To be entirely honestly, I don't think there's anything major that we've broke since the initial release, some minor features may have been dropped (in favor of plugins), but mostly we've added features, or improved options without breaking compatibility.

Backwards compatibility is extremely important for something that intends to last, such as a book format (like ours). It's highly unlikely that we break features before a `2.0.0` release (and even then we'll only do so where necessary).

As a testimony of backwards compatibility I tried building https://github.com/GitbookIO/javascript and https://github.com/GitbookIO/documentation with our `0.0.1` release (latest is `1.5.0`, 54 releases in total) 
 and they both work like a charm, maybe not as pretty and with less features but still work :) (no changes whatsoever ;) ).

That said if you want to build against a specific version of `gitbook`, you can use https://github.com/GitbookIO/grunt-gitbook.

We recommend to build against the latest release, and ideally using [GitBook.com](https://www.gitbook.com) where you get automated builds etc ... (it has great GitHub integration).

I don't think backwards compatibility is something you should be worried about, we've done a good job so far and I hope we'll continue to so in the future.

**Comment by sindresorhus** [comment]
To be honest, having to install `gitbook` globally is a no go for me. I know you think you won't break anything in the future, but the smallest differences in versions will be a problem for users, which might depend on some behaviour of a bug you fixed. Especially a problem in this case as books last for a long time. I was involved in grunt from the start, and we did that mistake and had to pay for it later in breaking all peoples projects.

What most build-systems like grunt/gulp/etc do is having the globally installed binary as just a dumb runner for the local core dependency. I would recommend you do the same.

You might be able to use https://github.com/tkellen/node-liftoff

> I don't think backwards compatibility is something you should be worried about, we've done a good job so far and I hope we'll continue to do so in the future.

Unfortunately you can't always control this.

-

These people can attest to the above:

// @tkellen @shama @contra @phated

**Comment by SamyPesse** [comment]
@sindresorhus I agree, you can install `gitbook` as a project dependency with a specific version (this is really not a problem). For example take a look at https://github.com/GitbookIO/grunt-gitbook (GitBook can be used from the command lien or programmatically).

**Comment by sindresorhus** [comment]
I don't want to use a grunt-plugin, but yeah, I could put it in package.json and do `npm run gitbook`. But would still be nicer to be able to do `gitbook` and run the local dependency. I also feel you should make this the default thing you recommend people use.

**Comment by SamyPesse** [comment]
The gitbook command line tool should check for a local version installed in the book folder and defaults to the version installed globally.

**Comment by sindresorhus** [comment]
No that's not what I'm asking. The `gitbook` binary should *only* check for a local version installed (not default). Otherwise, users *will* be lazy and just globally install without realizing the repercussions.

**Comment by phated** [comment]
:+1: to having a separate CLI. We are in the process of separating gulp's out into its own module.  There is one outstanding `npm` issue, but as soon as that is finished, we can ship a separate CLI *and* make it a dependency of a specific version of gulp.

**Comment by tkellen** [comment]
:+1: this will bite you eventually if it hasn't already

**Comment by jehoshua02** [comment]
... sound like this issue should be reopened.

**Comment by jehoshua02** [comment]
Another thing: you don't want fear of backward compatibility blocking you from making the product what it should be. Train your users to pay attention to versions, use semver, be careful, but don't fear.

**Comment by shama** [comment]
Luckily anything that can be installed globally with `npm` can also, by default, be installed locally.

So if anyone wants to write more than one book at a time they could do:
`npm i gitbook --save-dev` and `./node_modules/.bin/gitbook init`

Or more simply in the `package.json`:
```json
{
  "name": "mybook",
  "version": "1.0.0",
  "scripts": {
    "start": "gitbook serve ."
  },
  "devDependencies": {
    "gitbook": "^1.5.0"
  }
}
```

Then run `npm start`.

---

But :+1: for recommending the default instead of installing globally. Otherwise users will likely have a difficult time writing more than one book at a time if you decide to change the code here.

**Comment by mysticatea** [comment]
I try to use `gitbook` command in npm-scripts, but it's not working.
Cannot I use `gitbook` command in npm-scripts?

```json
{
  "private": true,
  "name": "my-docs",
  "version": "1.0.0",
  "description": "Documents",
  "scripts": {
    "build": "gitbook build",
    "watch": "gitbook serve"
  },
  "keywords": [],
  "author": "Toru Nagashima",
  "license": "MIT",
  "devDependencies": {
    "gitbook": "^2.6.7",
    "gitbook-cli": "^1.0.1",
    "gitbook-plugin-puml": "^1.0.1"
  }
}
```

```
$ npm install
...
$ npm run build
You need to install "gitbook-cli" to have access to the gitbook command anywhere on your system.
If you've installed this package globally, you need to uninstall it.
>> Run "npm uninstall -g gitbook" then "npm install -g gitbook-cli"
```

**Comment by SamyPesse** [comment]
Fixed in favour of #1377
